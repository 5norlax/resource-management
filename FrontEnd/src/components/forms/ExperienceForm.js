import React from 'react';
import { Label, Input, FormGroup, Row, Button, CustomInput } from 'reactstrap';

const ExperienceForm = ({ index, removeExperience, JobTitle, JobOrg, JobStartDate, JobEndDate, JobInfo, handleExperienceChange }) => {
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="JobTitle">Title</Label>
                    <Input required type="text" name="JobTitle" id="JobTitle" placeholder="Title/Role" value={JobTitle} onChange={(e) => { handleExperienceChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="JobOrg">Association</Label>
                    <Input required type="text" name="JobOrg" id="JobOrg" placeholder="Experience Association" value={JobOrg} onChange={(e) => { handleExperienceChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-3 col-lg-3 col-md-3 col-sm-6 col-xs-6">
                    <Label for="JobStartDate">Start Date</Label>
                    <Input
                        required
                        type="date"
                        name="JobStartDate"
                        id="JobStartDate"
                        value={JobStartDate}
                        placeholder="date placeholder"
                        onChange={(e) => { handleExperienceChange(index, e) }}
                    />
                </FormGroup>
                { index !== 0 ?
                <FormGroup className="col-xl-3 col-lg-3 col-md-3 col-sm-6 col-xs-6">
                    <Label for="JobStartDate">End Date</Label>
                    <Input
                        required
                        type="date"
                        name="JobEndDate"
                        id="JobEndDate"
                        value={JobEndDate}
                        placeholder="date placeholder"
                        onChange={(e) => { handleExperienceChange(index, e) }}
                    />
                </FormGroup>
                :
                <FormGroup style={{ marginTop: '40px' }} className="col-xl-3 col-lg-3 col-md-3 col-sm-6 col-xs-6">
                    <CustomInput type="checkbox" id="exampleCustomCheckbox" disabled checked label="Current Position" />
                </FormGroup>
                }
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="JobInfo">Details</Label>
                    <Input type="textarea" name="JobInfo" id="JobInfo" placeholder="Experience Details" value={JobInfo} onChange={(e) => { handleExperienceChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => { removeExperience(index) }} style={{ marginTop: '32px' }} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default ExperienceForm