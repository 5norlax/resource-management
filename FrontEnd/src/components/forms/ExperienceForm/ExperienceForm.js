import React, { useState } from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

export default function ExperienceForm({ index, removeExperience, JobTitle, JobOrg, JobStartDate, JobEndDate, JobInfo, handleExperienceChange }) {

    const [currentPosition, setCurrentPostion] = useState( JobEndDate === '' ? true : false)

    return (
        <div>
            <Row>
                <div className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12">
                    <FormGroup check>
                        <Label check>
                            <Input type="checkbox" className="unselectable" checked={currentPosition} 
                            onChange={() => {setCurrentPostion(!currentPosition)}} value={currentPosition}/>{' '}
                            Current Position
                        </Label>
                    </FormGroup>
                </div>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="JobTitle">Title</Label>
                    <Input required type="text" name="JobTitle" id="JobTitle" placeholder="Title/Role" value={JobTitle} onChange={(e) => { handleExperienceChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="JobOrg">Company/Organization</Label>
                    <Input required type="text" name="JobOrg" id="JobOrg" placeholder="Experience Association" value={JobOrg} onChange={(e) => { handleExperienceChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-3 col-lg-3 col-md-3 col-sm-6 col-xs-6">
                    <Label for="JobStartDate">Start Date</Label>
                    <Input
                        type="date"
                        name="JobStartDate"
                        id="JobStartDate"
                        value={JobStartDate}
                        placeholder="date placeholder"
                        onChange={(e) => { handleExperienceChange(index, e) }}
                    />
                </FormGroup>
                { currentPosition === true ?
                <FormGroup className="col-xl-3 col-lg-3 col-md-3 col-sm-6 col-xs-6">
                    <Label for="JobStartDate">End Date</Label>
                    <Input
                        disabled
                        type="date"
                        name="JobEndDate"
                        id="JobEndDate"
                        value={JobEndDate}
                        placeholder="date placeholder"
                        onChange={(e) => { handleExperienceChange(index, e) }}
                    />
                </FormGroup>
                :
                <FormGroup className="col-xl-3 col-lg-3 col-md-3 col-sm-6 col-xs-6">
                    <Label for="JobStartDate">End Date</Label>
                    <Input
                        required
                        type="date"
                        name="JobEndDate"
                        id="JobEndDate"
                        value={JobEndDate}
                        placeholder="date placeholder"
                        onChange={(e) => { handleExperienceChange(index, e) }}
                    />
                </FormGroup>
                }
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="JobInfo">Details</Label>
                    <Input type="textarea" name="JobInfo" id="JobInfo" placeholder="Experience Details" value={JobInfo} onChange={(e) => { handleExperienceChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => { removeExperience(index) }} style={{ marginTop: '32px' }} color="danger">Remove Experience</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
};