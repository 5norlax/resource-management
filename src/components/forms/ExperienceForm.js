import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

const ExperienceForm = ({index, removeExperience}) => {
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="expTitle">Title</Label>
                    <Input type="text" name="expName" id="expName" placeholder="Title/Role" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="expAssociation">Association</Label>
                    <Input type="text" name="expAssociation" id="expAssociation" placeholder="Experience Association" />
                </FormGroup>
                <FormGroup className="col-xl-2 col-lg-2 col-md-2 col-sm-4 col-xs-4">
                    <Label for="exStartpDate">Start Date</Label>
                    <Input type="text" name="exStartpDate" id="exStartpDate" placeholder="Start Date" />
                </FormGroup>
                <FormGroup className="col-xl-2 col-lg-2 col-md-2 col-sm-4 col-xs-4">
                    <Label for="expEndDate">End Date</Label>
                    <Input type="text" name="expEndDate" id="expEndDate" placeholder="End Date" />
                </FormGroup>
                <FormGroup check className="col-xl-2 col-lg-2 col-md-2 col-sm-4 col-xs-4">
                    <Label style={{ marginTop: '35px' }} check >
                    <Input type="checkbox" />{' '}
                        Current Position
                    </Label>
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="expDetails1">Details</Label>
                    <Input type="textarea" name="expDetails1" id="expDetails1" placeholder="Experience Details" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="expDetails2">Details</Label>
                    <Input type="textarea" name="expDetails2" id="expDetails2" placeholder="Experience Details" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="expDetails3">Details</Label>
                    <Input type="textarea" name="expDetails3" id="expDetails3" placeholder="Experience Details" />
                </FormGroup>
                <FormGroup className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => {removeExperience(index)}} style={{ marginTop: '32px' }} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default ExperienceForm