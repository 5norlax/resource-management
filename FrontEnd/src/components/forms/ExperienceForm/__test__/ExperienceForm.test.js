import React from 'react'
import ReactDOM from 'react-dom'
import { render, cleanup, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ExperienceForm from '../ExperienceForm'

afterEach(cleanup)

it('matches snapshot', async () => {
  const tree = renderer.create(<ExperienceForm />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders without crashing', async () => {
  const div = document.createElement('div')
  ReactDOM.render(<ExperienceForm />, div)
  ReactDOM.unmountComponentAtNode(div)
})

describe('Input Fields', () => {
  const handleExperienceChange = jest.fn()
  it('Current Position Input triggers handleExperienceChange onChange', async () => {
    const { queryByTestId } = render(<ExperienceForm JobEndDate={''} />)
    fireEvent.click(queryByTestId('currentPositionCheck'))
    expect(queryByTestId('currentPositionCheck')).toBeFalsy
  })
  it('Title Input triggers handleExperienceChange onChange', async () => {
    const { queryByPlaceholderText } = render(<ExperienceForm handleExperienceChange={handleExperienceChange} />)
    const titleInput = queryByPlaceholderText('Title/Role')
    fireEvent.change(titleInput, { target: { value: 'test' } })
    expect(handleExperienceChange).toHaveBeenCalled()
  })
  it('Company Input triggers handleExperienceChange onChange', async () => {
    const { queryByPlaceholderText } = render(<ExperienceForm handleExperienceChange={handleExperienceChange} />)
    const companyInput = queryByPlaceholderText('Experience Association')
    fireEvent.change(companyInput, { target: { value: 'test' } })
    expect(handleExperienceChange).toHaveBeenCalled()
  })
  it('Start Date Input triggers handleExperienceChange onChange', async () => {
    const { queryByTestId } = render(<ExperienceForm handleExperienceChange={handleExperienceChange} />)
    const dateInput = queryByTestId('startDate')
    fireEvent.change(dateInput, { target: { value: '0001-01-01' } })
    expect(handleExperienceChange).toHaveBeenCalled()
  })
  it('End Date Input triggers handleExperienceChange onChange', async () => {
    const { queryByTestId } = render(<ExperienceForm handleExperienceChange={handleExperienceChange} />)
    const dateInput = queryByTestId('endDate')
    fireEvent.change(dateInput, { target: { value: '0001-01-01' } })
    expect(handleExperienceChange).toHaveBeenCalled()
  })
  it('Details Input triggers handleExperienceChange onChange', async () => {
    const { queryByPlaceholderText } = render(<ExperienceForm handleExperienceChange={handleExperienceChange} />)
    const detailsInput = queryByPlaceholderText('Experience Details')
    fireEvent.change(detailsInput, { target: { value: 'test' } })
    expect(handleExperienceChange).toHaveBeenCalled()
  })
})

describe('Remove Button', () => {
  it('Button Removes Experience onClick', async () => {
    const removeExperience = jest.fn()
    const { queryByText } = render(<ExperienceForm removeExperience={removeExperience} />)
    fireEvent.click(queryByText('Remove Experience'))
    expect(removeExperience).toHaveBeenCalled()
  })
})