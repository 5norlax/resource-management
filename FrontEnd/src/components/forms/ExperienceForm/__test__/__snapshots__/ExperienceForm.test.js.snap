// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12"
    >
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="unselectable form-check-input"
            data-testid="currentPositionCheck"
            onChange={[Function]}
            type="checkbox"
            value={false}
          />
           
          Current Position
        </label>
      </div>
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="JobTitle"
      >
        Title
      </label>
      <input
        className="form-control"
        id="JobTitle"
        name="JobTitle"
        onChange={[Function]}
        placeholder="Title/Role"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="JobOrg"
      >
        Company/Organization
      </label>
      <input
        className="form-control"
        id="JobOrg"
        name="JobOrg"
        onChange={[Function]}
        placeholder="Experience Association"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-3 col-lg-3 col-md-3 col-sm-6 col-xs-6 form-group"
    >
      <label
        className=""
        htmlFor="JobStartDate"
      >
        Start Date
      </label>
      <input
        className="form-control"
        data-testid="startDate"
        id="JobStartDate"
        name="JobStartDate"
        onChange={[Function]}
        placeholder="date placeholder"
        type="date"
      />
    </div>
    <div
      className="col-xl-3 col-lg-3 col-md-3 col-sm-6 col-xs-6 form-group"
    >
      <label
        className=""
        htmlFor="JobStartDate"
      >
        End Date
      </label>
      <input
        className="form-control"
        data-testid="endDate"
        id="JobEndDate"
        name="JobEndDate"
        onChange={[Function]}
        placeholder="date placeholder"
        required={true}
        type="date"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="JobInfo"
      >
        Details
      </label>
      <textarea
        className="form-control"
        id="JobInfo"
        name="JobInfo"
        onChange={[Function]}
        placeholder="Experience Details"
      />
    </div>
    <div
      className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12 text-right form-group"
    >
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        style={
          Object {
            "marginTop": "32px",
          }
        }
        type="button"
      >
        Remove Experience
      </button>
    </div>
  </div>
  <hr />
</div>
`;
